pkg/connection/interfaces.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package connection
```

Comment:
`connection` might not be a great package name. I wanted to say `connectiondetails` but chose `connection` for brevity. However, now I am thinking that it could be confusing as if `connection.Manager` is managing some connections. 
One other alternative was `secret` which sounds better now. Open to other alternatives.

Reply:
Hrm, I like `connection` in most cases - e.g. `connection.DetailsPublisher`, `connection.SecretOwner` etc all read well. Could you address the `Manager` concern by renaming it to `connection.DetailsManager`?

Reply:
Hmm, yeah, sounds like a good idea. 
Checking again `Manager` seems to be the only one that could be confusing.